AstSemantics.md

```diff
@@ -316,14 +283,11 @@ in the function table.
 
 Function-pointer values are comparable for equality and the `addressof` operator
 is monomorphic. Function-pointer values can be explicitly coerced to and from
-integers (which, in particular, is necessary when loading/storing to memory
-since memory only provides integer types). For security and safety reasons,
-the integer value of a coerced function-pointer value is an abstract index and
-does not reveal the actual machine code address of the target function.
+integers.
```

Comment:
This section will conflict with https://github.com/WebAssembly/design/pull/392. Maybe just leave as is?